src/mac.rs

```diff
@@ -514,64 +525,165 @@ impl Security {
-///
-/// Part of [`Header`]. PAN ID compression is currently not supported by this
-/// implementation.
+/// Defines version information for a frame
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub enum PanIdCompress {
-    /// PAN ID compression is disabled
-    Disabled = 0b0,
+pub enum FrameVersion {
+    /// A frame conforming to the 802.15.4-2003 standard
+    Ieee802154_2003 = 0b00,
+    /// A frame conforming to the 802.15.4-2006 standard
+    Ieee802154_2006 = 0b01,
+    /// A frame conforming to the current 802.15.4 standard
+    Ieee802154 = 0b10,
 }
 
-impl PanIdCompress {
-    /// Creates an instance of `PanIdCompress` from the provided bits
+impl FrameVersion {
+    /// Creates an instance of [`FrameVersion`] from the provided bits
     ///
     /// Returns `None`, if the provided bits don't encode a valid value of
-    /// `PanIdCompress`.
+    /// `FrameVersion`.
     ///
     /// # Example
     ///
     /// ``` rust
-    /// use ieee802154::mac::PanIdCompress;
+    /// use ieee802154::mac::FrameVersion;
     ///
-    /// let pan_id_compress = PanIdCompress::from_bits(0b0);
-    /// assert_eq!(pan_id_compress, Some(PanIdCompress::Disabled));
+    /// let version = FrameVersion::from_bits(0b0);
+    /// assert_eq!(version, Some(FrameVersion::Ieee802154_2003));
     /// ```
     pub fn from_bits(bits: u8) -> Option<Self> {
         match bits {
-            0b0 => Some(PanIdCompress::Disabled),
-            _   => None,
+            0b00 => Some(FrameVersion::Ieee802154_2003),
+            0b01 => Some(FrameVersion::Ieee802154_2006),
+            0b10 => Some(FrameVersion::Ieee802154),
+            _ => None,
         }
     }
 }
 
+/// Defines the type of Address
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum AddressMode {
+    /// PAN identifier and address field are not present
+    None = 0b00,
+    /// Address field contains a 16 bit short address
+    Short = 0b10,
+    /// Address field contains a 64 bit extended address
+    Extended = 0b11,
+}
 
-/// An address consisting of PAN ID and short address
+impl AddressMode {
+    /// Creates an instance of [`AddressMode`] from the provided bits
+    ///
+    /// Returns `None`, if the provided bits don't encode a valid value of
+    /// `AddressMode`.
+    ///
+    /// # Example
+    ///
+    /// ``` rust
+    /// use ieee802154::mac::AddressMode;
+    ///
+    /// let address_mode = AddressMode::from_bits(0b10).unwrap();
+    /// assert_eq!(address_mode, AddressMode::Short);
+    /// ```
+    pub fn from_bits(bits: u8) -> Result<Self, DecodeError> {
+        match bits {
+            0b00 => Ok(AddressMode::None),
+            0b10 => Ok(AddressMode::Short),
+            0b11 => Ok(AddressMode::Extended),
+            _ => Err(DecodeError::InvalidAddressMode(bits)),
+        }
+    }
+}
+
+/// Personal Area Network Identifier
 ///
-/// Other address variants are currently not supported by this implementation.
+/// A 16-bit identifier for devices in a PAN
```

Comment:
The identifier is for the PAN itself, not devices within it, right?

Reply:
Correct.